This was in the old xim-fixes3.diff before cws vcl36 was merged. vcl36
code doesn't have the m_bWasPreedit condition. I don't know if it
prevents double commits in another way or if it uses m_bWasPreedit
with semantics that differ from xim-fixes3.diff
The IZ entry was #38671 -- martink 2005-03-07

Index: vcl/unx/gtk/window/gtkframe.cxx
===================================================================
RCS file: /cvs/gsl/vcl/unx/gtk/window/gtkframe.cxx,v
retrieving revision 1.17
diff -u -r1.17 gtkframe.cxx
--- vcl/unx/gtk/window/gtkframe.cxx	16 Nov 2004 15:19:08 -0000	1.17
+++ vcl/unx/gtk/window/gtkframe.cxx	15 Dec 2004 15:57:34 -0000
@@ -1971,7 +2031,20 @@ void GtkSalFrame::signalIMPreeditEnd( GtkIMContext* pContext, gpointer frame )
 
 void GtkSalFrame::signalIMPreeditEnd( GtkIMContext* pContext, gpointer frame )
 {
-//    GtkSalFrame* pThis = (GtkSalFrame*)frame;
+    GtkSalFrame* pThis = (GtkSalFrame*)frame;
+#if OSL_DEBUG_LEVEL > 1
+	fprintf( stderr, "*** :signalImPreeditEnd *** '%s'\n", pThis->m_bWasPreedit ? "commit" : "nop" );
+#endif
+    GTK_YIELD_GRAB();
+	if( pThis->m_bWasPreedit )
+	{
+		pThis->m_bWasPreedit = false;
+		pThis->CallCallback( SALEVENT_ENDEXTTEXTINPUT, NULL );
+	}
+#if OSL_DEBUG_LEVEL > 1
+	else
+		fprintf (stderr, "Avoid double commit\n");
+#endif
 }
 
 gboolean GtkSalFrame::signalIMRetrieveSurrounding( GtkIMContext* pContext, gpointer frame )
